// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/DocumentListUsage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDocumentListUsage = instanceOfDocumentListUsage;
exports.DocumentListUsageFromJSON = DocumentListUsageFromJSON;
exports.DocumentListUsageFromJSONTyped = DocumentListUsageFromJSONTyped;
exports.DocumentListUsageToJSON = DocumentListUsageToJSON;
/**
 * Check if a given object implements the DocumentListUsage interface.
 */
function instanceOfDocumentListUsage(value) {
    let isInstance = true;
    isInstance = isInstance && "readUnits" in value;
    return isInstance;
}
function DocumentListUsageFromJSON(json) {
    return DocumentListUsageFromJSONTyped(json, false);
}
function DocumentListUsageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'readUnits': json['read_units'],
    };
}
function DocumentListUsageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'read_units': value.readUnits,
    };
}
