// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/ListedDocumentRecord.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListedDocumentRecord = instanceOfListedDocumentRecord;
exports.ListedDocumentRecordFromJSON = ListedDocumentRecordFromJSON;
exports.ListedDocumentRecordFromJSONTyped = ListedDocumentRecordFromJSONTyped;
exports.ListedDocumentRecordToJSON = ListedDocumentRecordToJSON;
/**
 * Check if a given object implements the ListedDocumentRecord interface.
 */
function instanceOfListedDocumentRecord(value) {
    let isInstance = true;
    isInstance = isInstance && "_id" in value;
    return isInstance;
}
function ListedDocumentRecordFromJSON(json) {
    return ListedDocumentRecordFromJSONTyped(json, false);
}
function ListedDocumentRecordFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        '_id': json['_id'],
    };
}
function ListedDocumentRecordToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '_id': value._id,
    };
}
