// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/ListDocumentsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListDocumentsResponse = instanceOfListDocumentsResponse;
exports.ListDocumentsResponseFromJSON = ListDocumentsResponseFromJSON;
exports.ListDocumentsResponseFromJSONTyped = ListDocumentsResponseFromJSONTyped;
exports.ListDocumentsResponseToJSON = ListDocumentsResponseToJSON;
const runtime_1 = require("../runtime");
const DocumentListUsage_1 = require("./DocumentListUsage");
const ListedDocumentRecord_1 = require("./ListedDocumentRecord");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the ListDocumentsResponse interface.
 */
function instanceOfListDocumentsResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "documents" in value;
    isInstance = isInstance && "namespace" in value;
    isInstance = isInstance && "usage" in value;
    return isInstance;
}
function ListDocumentsResponseFromJSON(json) {
    return ListDocumentsResponseFromJSONTyped(json, false);
}
function ListDocumentsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'documents': (json['documents'].map(ListedDocumentRecord_1.ListedDocumentRecordFromJSON)),
        'pagination': !(0, runtime_1.exists)(json, 'pagination') ? undefined : (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'namespace': json['namespace'],
        'usage': (0, DocumentListUsage_1.DocumentListUsageFromJSON)(json['usage']),
    };
}
function ListDocumentsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'documents': (value.documents.map(ListedDocumentRecord_1.ListedDocumentRecordToJSON)),
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'namespace': value.namespace,
        'usage': (0, DocumentListUsage_1.DocumentListUsageToJSON)(value.usage),
    };
}
