// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/ListDocumentsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListDocumentsRequest = instanceOfListDocumentsRequest;
exports.ListDocumentsRequestFromJSON = ListDocumentsRequestFromJSON;
exports.ListDocumentsRequestFromJSONTyped = ListDocumentsRequestFromJSONTyped;
exports.ListDocumentsRequestToJSON = ListDocumentsRequestToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ListDocumentsRequest interface.
 */
function instanceOfListDocumentsRequest(value) {
    let isInstance = true;
    return isInstance;
}
function ListDocumentsRequestFromJSON(json) {
    return ListDocumentsRequestFromJSONTyped(json, false);
}
function ListDocumentsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'prefix': !(0, runtime_1.exists)(json, 'prefix') ? undefined : json['prefix'],
        'limit': !(0, runtime_1.exists)(json, 'limit') ? undefined : json['limit'],
        'paginationToken': !(0, runtime_1.exists)(json, 'pagination_token') ? undefined : json['pagination_token'],
    };
}
function ListDocumentsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'prefix': value.prefix,
        'limit': value.limit,
        'pagination_token': value.paginationToken,
    };
}
