// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/UpdateDocumentRecord.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateDocumentRecord = instanceOfUpdateDocumentRecord;
exports.UpdateDocumentRecordFromJSON = UpdateDocumentRecordFromJSON;
exports.UpdateDocumentRecordFromJSONTyped = UpdateDocumentRecordFromJSONTyped;
exports.UpdateDocumentRecordToJSON = UpdateDocumentRecordToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateDocumentRecord interface.
 */
function instanceOfUpdateDocumentRecord(value) {
    let isInstance = true;
    isInstance = isInstance && "_id" in value;
    return isInstance;
}
function UpdateDocumentRecordFromJSON(json) {
    return UpdateDocumentRecordFromJSONTyped(json, false);
}
function UpdateDocumentRecordFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        ...json,
        '_id': json['_id'],
        '_remove_fields': !(0, runtime_1.exists)(json, '_remove_fields') ? undefined : json['_remove_fields'],
    };
}
function UpdateDocumentRecordToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        ...value,
        '_id': value._id,
        '_remove_fields': value._remove_fields,
    };
}
