// @pinecone-database/pinecone/dist/preview/indexes/__tests__/listDocuments.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const listDocuments_1 = require("../listDocuments");
const errors_1 = require("../../../errors");
const mockResponse = {
    documents: [{ _id: 'doc-1' }, { _id: 'doc-2' }],
    pagination: { next: 'next-token' },
    namespace: 'test-ns',
    usage: { readUnits: 1 },
};
const buildMockApi = (overrides = {}) => ({
    listDocuments: jest.fn().mockResolvedValue(mockResponse),
    ...overrides,
});
describe('listPreviewDocuments', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when limit is less than 1', async () => {
            await expect((0, listDocuments_1.listPreviewDocuments)(buildMockApi(), 'test-ns', { limit: 0 })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references `limit` when it is invalid', async () => {
            await expect((0, listDocuments_1.listPreviewDocuments)(buildMockApi(), 'test-ns', { limit: -5 })).rejects.toThrow('`limit`');
        });
        test('does not throw when no options are provided', async () => {
            await expect((0, listDocuments_1.listPreviewDocuments)(buildMockApi(), 'test-ns')).resolves.toBeDefined();
        });
    });
    describe('success', () => {
        test('returns PreviewListDocumentsResponse with documents, namespace, usage', async () => {
            const result = await (0, listDocuments_1.listPreviewDocuments)(buildMockApi(), 'test-ns');
            expect(result.documents).toBeDefined();
            expect(result.namespace).toBe('test-ns');
            expect(result.usage.readUnits).toBeGreaterThanOrEqual(0);
        });
        test('passes the pagination token through on the response', async () => {
            const result = await (0, listDocuments_1.listPreviewDocuments)(buildMockApi(), 'test-ns');
            expect(result.pagination?.next).toBe('next-token');
        });
        test('passes namespace to the API call', async () => {
            const api = buildMockApi();
            await (0, listDocuments_1.listPreviewDocuments)(api, 'my-namespace');
            expect(api.listDocuments).toHaveBeenCalledWith(expect.objectContaining({ namespace: 'my-namespace' }));
        });
        test('passes prefix, limit, and paginationToken to the API call', async () => {
            const api = buildMockApi();
            const options = {
                prefix: 'doc-',
                limit: 50,
                paginationToken: 'token-abc',
            };
            await (0, listDocuments_1.listPreviewDocuments)(api, 'test-ns', options);
            expect(api.listDocuments).toHaveBeenCalledWith(expect.objectContaining({
                listDocumentsRequest: expect.objectContaining({
                    prefix: 'doc-',
                    limit: 50,
                    paginationToken: 'token-abc',
                }),
            }));
        });
        test('passes xPineconeApiVersion containing 2026 in every call', async () => {
            const api = buildMockApi();
            await (0, listDocuments_1.listPreviewDocuments)(api, 'test-ns');
            expect(api.listDocuments).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('converts ResponseError (404) to a typed Pinecone error via handleApiError', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_data')));
            const api = buildMockApi({
                listDocuments: jest.fn().mockRejectedValue(new ResponseError(new Response('{"code": 5, "message": "Index not found"}', {
                    status: 404,
                }), 'Not Found')),
            });
            await expect((0, listDocuments_1.listPreviewDocuments)(api, 'test-ns')).rejects.toBeDefined();
        });
    });
});
