// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/ResponseStringFieldFullTextSearchNgram.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfResponseStringFieldFullTextSearchNgram = instanceOfResponseStringFieldFullTextSearchNgram;
exports.ResponseStringFieldFullTextSearchNgramFromJSON = ResponseStringFieldFullTextSearchNgramFromJSON;
exports.ResponseStringFieldFullTextSearchNgramFromJSONTyped = ResponseStringFieldFullTextSearchNgramFromJSONTyped;
exports.ResponseStringFieldFullTextSearchNgramToJSON = ResponseStringFieldFullTextSearchNgramToJSON;
/**
 * Check if a given object implements the ResponseStringFieldFullTextSearchNgram interface.
 */
function instanceOfResponseStringFieldFullTextSearchNgram(value) {
    let isInstance = true;
    isInstance = isInstance && "minGram" in value;
    isInstance = isInstance && "maxGram" in value;
    isInstance = isInstance && "prefixOnly" in value;
    return isInstance;
}
function ResponseStringFieldFullTextSearchNgramFromJSON(json) {
    return ResponseStringFieldFullTextSearchNgramFromJSONTyped(json, false);
}
function ResponseStringFieldFullTextSearchNgramFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'minGram': json['min_gram'],
        'maxGram': json['max_gram'],
        'prefixOnly': json['prefix_only'],
    };
}
function ResponseStringFieldFullTextSearchNgramToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'min_gram': value.minGram,
        'max_gram': value.maxGram,
        'prefix_only': value.prefixOnly,
    };
}
