// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/StringFieldFullTextSearchNgram.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStringFieldFullTextSearchNgram = instanceOfStringFieldFullTextSearchNgram;
exports.StringFieldFullTextSearchNgramFromJSON = StringFieldFullTextSearchNgramFromJSON;
exports.StringFieldFullTextSearchNgramFromJSONTyped = StringFieldFullTextSearchNgramFromJSONTyped;
exports.StringFieldFullTextSearchNgramToJSON = StringFieldFullTextSearchNgramToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the StringFieldFullTextSearchNgram interface.
 */
function instanceOfStringFieldFullTextSearchNgram(value) {
    let isInstance = true;
    isInstance = isInstance && "minGram" in value;
    isInstance = isInstance && "maxGram" in value;
    return isInstance;
}
function StringFieldFullTextSearchNgramFromJSON(json) {
    return StringFieldFullTextSearchNgramFromJSONTyped(json, false);
}
function StringFieldFullTextSearchNgramFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'minGram': json['min_gram'],
        'maxGram': json['max_gram'],
        'prefixOnly': !(0, runtime_1.exists)(json, 'prefix_only') ? undefined : json['prefix_only'],
    };
}
function StringFieldFullTextSearchNgramToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'min_gram': value.minGram,
        'max_gram': value.maxGram,
        'prefix_only': value.prefixOnly,
    };
}
