// @pinecone-database/pinecone/dist/integration/preview-data/updateDocuments.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const index_1 = require("../../index");
const test_helpers_1 = require("../test-helpers");
let pc;
let indexName;
const namespace = 'test-ns';
beforeAll(async () => {
    pc = new index_1.Pinecone();
    indexName = (0, test_helpers_1.randomName)('preview-update-docs');
    await pc.preview.indexes.create({
        name: indexName,
        schema: {
            fields: {
                content: { type: 'string', fullTextSearch: {} },
            },
        },
        waitUntilReady: true,
    });
    await pc.preview.index({ name: indexName }).upsertDocuments(namespace, {
        documents: [
            { _id: 'doc-1', title: 'Original title', content: 'Machine learning' },
        ],
    });
    // Poll until the upserted document is retrievable
    await (0, test_helpers_1.assertWithRetries)(() => pc.preview
        .index({ name: indexName })
        .fetchDocuments(namespace, { ids: ['doc-1'] }), (result) => {
        expect(result.documents['doc-1']).toBeDefined();
    });
}, 300000);
afterAll(async () => {
    try {
        await pc.preview.indexes.delete(indexName);
    }
    finally {
        // cleanup attempt completed
    }
});
describe('preview updateDocuments', () => {
    test('applies a partial field update and removes a field', async () => {
        await pc.preview.index({ name: indexName }).updateDocuments(namespace, {
            documents: [
                { _id: 'doc-1', content: 'Updated content', _remove_fields: ['title'] },
            ],
        });
        // Poll until the update has propagated
        await (0, test_helpers_1.assertWithRetries)(() => pc.preview
            .index({ name: indexName })
            .fetchDocuments(namespace, { ids: ['doc-1'] }), (result) => {
            const doc = result.documents['doc-1'];
            expect(doc).toBeDefined();
            expect(doc.content).toBe('Updated content');
            expect(doc.title).toBeUndefined();
        });
    });
});
